src/main/java/net/dv8tion/jda/core/requests/restaction/GuildAction.java

```diff
@@ -0,0 +1,733 @@
+     *         The {@link net.dv8tion.jda.core.entities.Guild.VerificationLevel VerificationLevel} to use
+     *
+     * @return The current GuildAction for chaining convenience
+     */
+    @CheckReturnValue
+    public GuildAction setVerificationLevel(Guild.VerificationLevel level)
+    {
+        this.verificationLevel = level;
+        return this;
+    }
+
+    /**
+     * Sets the {@link net.dv8tion.jda.core.entities.Guild.NotificationLevel NotificationLevel}
+     * for the resulting {@link net.dv8tion.jda.core.entities.Guild Guild}
+     *
+     * @param  level
+     *         The {@link net.dv8tion.jda.core.entities.Guild.NotificationLevel NotificationLevel} to use
+     *
+     * @return The current GuildAction for chaining convenience
+     */
+    @CheckReturnValue
+    public GuildAction setNotificationLevel(Guild.NotificationLevel level)
+    {
+        this.notificationLevel = level;
+        return this;
+    }
+
+    /**
+     * Sets the {@link net.dv8tion.jda.core.entities.Guild.ExplicitContentLevel ExplicitContentLevel}
+     * for the resulting {@link net.dv8tion.jda.core.entities.Guild Guild}
+     *
+     * @param  level
+     *         The {@link net.dv8tion.jda.core.entities.Guild.ExplicitContentLevel ExplicitContentLevel} to use
+     *
+     * @return The current GuildAction for chaining convenience
+     */
+    @CheckReturnValue
+    public GuildAction setExplicitContentLevel(Guild.ExplicitContentLevel level)
+    {
+        this.explicitContentLevel = level;
+        return this;
+    }
+
+    // Channels
+
+    /**
+     * Adds a {@link net.dv8tion.jda.core.entities.Channel Channel} to the resulting
+     * Guild. This cannot be of type {@link net.dv8tion.jda.core.entities.ChannelType#CATEGORY CATEGORY}!
+     *
+     * @param  channel
+     *         The {@link net.dv8tion.jda.core.requests.restaction.GuildAction.ChannelData ChannelData}
+     *         to use for the construction of the Channel
+     *
+     * @throws java.lang.IllegalArgumentException
+     *         If the provided channel is {@code null}!
+     *
+     * @return The current GuildAction for chaining convenience
+     */
+    @CheckReturnValue
+    public GuildAction addChannel(ChannelData channel)
+    {
+        Checks.notNull(channel, "Channel");
+        this.channels.add(channel);
+        return this;
+    }
+
+    /**
+     * Gets the {@link net.dv8tion.jda.core.requests.restaction.GuildAction.ChannelData ChannelData}
+     * of the specified index. The index is 0 based on insertion order of {@link #addChannel(GuildAction.ChannelData)}!
+     *
+     * @param  index
+     *         The 0 based index of the channel
+     *
+     * @throws java.lang.IndexOutOfBoundsException
+     *         If the provided index is not in bounds
+     *
+     * @return The current GuildAction for chaining convenience
+     */
+    @CheckReturnValue
+    public ChannelData getChannel(int index)
+    {
+        return this.channels.get(index);
+    }
```

Comment:
If you are supporting `addChannel` and `getChannel`, probably need to support `removeChannel`